numpy/testing/_private/utils.py

```diff
@@ -780,10 +783,10 @@ def func_assert_same_pos(x, y, func=isnan, hasval='nan'):
         # do not trigger a failure (np.ma.masked != True evaluates as
         # np.ma.masked, which is falsy).
         if cond != True:
-            match = 100-100.0*reduced.count(1)/len(reduced)
-            msg = build_err_msg([x, y],
+            mismatch = 100.0 * reduced.count(0) / ox.size
+            msg = build_err_msg([ox, oy],
                                 err_msg
-                                + '\n(mismatch %s%%)' % (match,),
+                                + '\n(mismatch %s%%)' % (mismatch,),
```

Comment:
Why did you change the calculation here? Can you show this is more correct?

Reply:
Sure. First of all, the variable name `match` is misleading, because what it actually means is `mismatch`, which you can tell from the string the variable tries to format.
The original way to calculate `match` (which is actually mismatch) is firstly calculate match percentage and then subtract it from 100. This logic is verbose and not compatible with this bug fix, because in this bug fix we are trying to count NaN's as matched result, which is not taken into account in the NaN-free `reduced`. So here it's not feasible to count results that matched (`1`), however results that are not matched (`0`) are stil meaningful here.